fgpyo/platform/illumina.py

```diff
@@ -0,0 +1,65 @@
+from enum import Enum
+from enum import unique
+from glob import glob
+from pathlib import Path
+from typing import List
+
+
+@unique
+class IlluminaReadNumber(Enum):
+    """The read number of an Illumina FASTQ."""
+
+    R1 = "R1"
+    R2 = "R2"
+    I1 = "I1"
+    I2 = "I2"
+
+
+FASTQ_EXTENSIONS = [".fq", ".fastq", ".fq.gz", ".fastq.gz"]
+
+
+def find_illumina_fastq(
+    fastq_dir: Path,
+    sample_id: str,
+    read_number: IlluminaReadNumber,
+    recursive: bool = True,
+) -> Path:
```

Comment:
@nh13 Should this be updated to return `list[Path]`, such that if a sample was split across lanes the function returns all FASTQs from the sample?

Reply:
Yes please.  Also consider if the read is chunked into multiple parts due to file size, which I have seen.